src/VisualStudio/LiveShare/Impl/Client/LocalForwarders/CSharpRemoteBraceMatchers.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.CSharp.BraceMatching;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.VisualStudio.LanguageServices.LiveShare.Client.LocalForwarders
+{
+    [ExportBraceMatcher(StringConstants.CSharpLspLanguageName)]
+    internal class CSharpRemoteOpenCloseBraceBraceMatcher : OpenCloseBraceBraceMatcher
```

Comment:
i personally don't like having concrete exported types derive from concrete exported types (or concrete types deriving from concrete types in general).

Why do we have to do it this way?  Can we not just put both exports on the base type?

Reply:
The StringConstants is explictly in the LSP layer partly to keep that self contained....

Reply:
Although I'll admit in this case it is pretty ugly, since there's no easy way to confirm we got all of them here short of me looking at both files and making sure there's a 1:1 match.